### PR TITLE
Components that renders F24 on Notification Payment

### PR DESCRIPTION
**Pull request by alessandrogelmi**
## Short description
Components that renders F24 on Notification Payment

## List of changes proposed in this pull request
- Created component

### CODE REVIEW

File: packages/pn-commons/src/components/NotificationDetail/NotificationPaymentRecipient.tsx [hunk 1]
```diff
@@ -27,29 +33,61 @@ const NotificationPaymentRecipient: React.FC<Props> = ({
   handleDownloadAttachamentPagoPA,
   handleReloadPayment,
 }) => {
-  const isSinglePayment = payments.length === 1;
+  const { pagoPaF24, f24Only } = payments;
+
+  const isSinglePayment = pagoPaF24.length === 1;
 
   const [selectedPayment, setSelectedPayment] = useState<PagoPAPaymentHistory | null>(null);
```
Comment (AndreaCimini90) [inline packages/pn-commons/src/components/NotificationDetail/NotificationPaymentRecipient.tsx:40]: Now if you have the payments already set, you can do useState<PagoPAPaymentHistory | null>(isSinglePayment ? ... : null);

File: packages/pn-commons/src/types/NotificationDetail.ts [hunk 2]
```diff
@@ -34,7 +34,7 @@ export type PagoPAPaymentHistory = PagoPAPaymentDetails & PaidDetails & ExtRegis
 
 export interface PaymentHistory {
```
Comment (AndreaCimini90) [inline packages/pn-commons/src/types/NotificationDetail.ts:35]: Rename the model in PaymentDetails

File: packages/pn-commons/src/types/NotificationDetail.ts [hunk 3]
```diff
@@ -34,7 +34,7 @@ export type PagoPAPaymentHistory = PagoPAPaymentDetails & PaidDetails & ExtRegis
...
   pagoPA?: PagoPAPaymentHistory;
```
Comment (AndreaCimini90) [inline packages/pn-commons/src/types/NotificationDetail.ts:36]: Rename the model iin PagoPAPaymentFullDetails

File: packages/pn-commons/src/types/NotificationDetail.ts [hunk 4]
```diff
@@ -248,17 +251,18 @@ export interface PagoPAPaymentDetails {
...
-  applyCostFlg: boolean;
+  applyCost: boolean;
 }
 
 export interface F24PaymentDetails {
-  description: string;
-  metadata: any;
+  title: string;
+  applyCost: boolean;
+  metadataAttachment: any; // TODO replace any with Attachment type
```
Comment (AndreaCimini90) [inline packages/pn-commons/src/types/NotificationDetail.ts:260]: Do the TODO

File: packages/pn-commons/src/utils/notification.utility.ts [hunk 5]
```diff
@@ -855,40 +855,29 @@ const populateOtherDocuments = (
...
-
-  // 2. Get array of payments of user
-  const userPayments = user?.payments;
-
-  if (!userPayments || userPayments.length === 0) {
+  if (!pagoPaF24Payemnts || pagoPaF24Payemnts.length === 0) {
     return [];
   }
 
   // 3. Get all timeline steps that have category payment
```
Comment (AndreaCimini90) [inline packages/pn-commons/src/utils/notification.utility.ts:875]: change the numeration

File: packages/pn-pa-webapp/src/__mocks__/NotificationDetail.mock.ts [hunk 6]
```diff
@@ -450,21 +450,24 @@ const paymentsPagoPaF24: Array<NotificationDetailPayment> = [
         },
       },
     },
-    f24Data: {
-      metadata: 'metadata-mocked',
-      description: 'F24 prima rata TARI',
+    f24: {
+      metadataAttachment: 'metadata-mocked',
```
Comment (AndreaCimini90) [inline packages/pn-pa-webapp/src/__mocks__/NotificationDetail.mock.ts:454]: adapt the mock to the new model

File: packages/pn-personafisica-webapp/src/__mocks__/NotificationDetail.mock.ts [hunk 7]
```diff
@@ -52,23 +52,31 @@ const payments: Array<NotificationDetailPayment> = [
         },
       },
     },
-    f24Data: {
-      metadata: 'metadata-mocked',
-      description: 'F24 seconda rata TARI',
+    f24: {
+      metadataAttachment: 'metadata-mocked',
```
Comment (AndreaCimini90) [inline packages/pn-personafisica-webapp/src/__mocks__/NotificationDetail.mock.ts:56]: adapt to new model

File: packages/pn-personafisica-webapp/src/redux/notification/reducers.ts [hunk 8]
```diff
@@ -63,7 +63,11 @@ const initialState = {
   f24AttachmentUrl: '',
   downtimeLegalFactUrl: '', // the non-filled value for URLs must be a falsy value in order to ensure expected behavior of useDownloadDocument
   // analogous for other URLs
-  paymentInfo: [] as Array<PaymentHistory>,
+  // paymentInfo: [] as Array<PaymentHistory>,
```
Comment (AndreaCimini90) [inline packages/pn-personafisica-webapp/src/redux/notification/reducers.ts:66]: remove commented code

File: packages/pn-personafisica-webapp/src/redux/notification/reducers.ts [hunk 9]
```diff
@@ -80,12 +84,42 @@ const notificationSlice = createSlice({
...
-      state.paymentInfo = action.payload;
-    },
   },
   extraReducers: (builder) => {
     builder.addCase(getReceivedNotification.fulfilled, (state, action) => {
+      const paymentsOfRecipient = action.payload.recipients.find(
+        (recipient) => recipient.taxId === action.payload.currentRecipient.taxId
+      )?.payments;
+
+      if (paymentsOfRecipient) {
```
Comment (AndreaCimini90) [inline packages/pn-personafisica-webapp/src/redux/notification/reducers.ts:94]: maybe here you can create two functions in notification.utility.tsx. One that get the pagoPaF24 payments and one that get only the f24

File: packages/pn-personafisica-webapp/src/redux/notification/reducers.ts [hunk 10]
```diff
@@ -166,28 +200,28 @@ const notificationSlice = createSlice({
           payment.isLoading = true;
           return;
         }
-        state.paymentInfo = [{ ...paymentHistory[0], isLoading: true }];
+        state.paymentsData.pagoPaF24 = [{ ...state.paymentsData.pagoPaF24[0], isLoading: true }];
```
Comment (AndreaCimini90) [inline packages/pn-personafisica-webapp/src/redux/notification/reducers.ts:203]: Maybe this code is useless. try to remove and check that the code works

File: packages/pn-personafisica-webapp/src/redux/notification/reducers.ts [hunk 11]
```diff
@@ -120,43 +154,43 @@ const notificationSlice = createSlice({
...
       }
     });
     builder.addCase(getNotificationPaymentInfo.pending, (state, action) => {
-      const paymentHistory = populatePaymentHistory(
-        action.meta.arg.taxId,
-        state.notification.timeline,
-        state.notification.recipients,
-        action.meta.arg.paymentInfoRequest as Array<ExtRegistriesPaymentDetails>
-      );
+      // const paymentHistory = populatePaymentsPagoPaF24(
```
Comment (AndreaCimini90) [inline packages/pn-personafisica-webapp/src/redux/notification/reducers.ts:177]: remove commented code

File: packages/pn-personafisica-webapp/src/redux/notification/reducers.ts [hunk 12]
```diff
@@ -166,28 +200,28 @@ const notificationSlice = createSlice({
...
       const creditorTaxId = action.meta.arg.paymentNotice.fiscalCode;
-      const paymentInfo = state.paymentInfo.find(
+      const paymentInfo = state.paymentsData.pagoPaF24.find(
         (payment) =>
           payment.pagoPA?.creditorTaxId === creditorTaxId &&
           payment.pagoPA?.noticeCode === noticeCode
       );
 
       if (paymentInfo && paymentInfo.pagoPA) {
         const updatedPaymentInfo: PaymentHistory = {
```
Comment (AndreaCimini90) [inline packages/pn-personafisica-webapp/src/redux/notification/reducers.ts:216]: Also here, maybe you can update paymentInfo variable directly and remove the row 224.

File: packages/pn-personagiuridica-webapp/src/__mocks__/NotificationDetail.mock.ts [hunk 13]
```diff
@@ -50,23 +50,24 @@ const paymentsPagoPA: Array<NotificationDetailPayment> = [
         },
       },
     },
-    f24Data: {
-      metadata: 'metadata-mocked',
-      description: 'F24 seconda rata TARI',
+    f24: {
+      metadataAttachment: 'metadata-mocked',
```
Comment (AndreaCimini90) [inline packages/pn-personagiuridica-webapp/src/__mocks__/NotificationDetail.mock.ts:54]: Adapt to new model

**Review by AndreaCimini90 — CHANGES_REQUESTED** [review]

File: packages/pn-personafisica-webapp/src/redux/notification/reducers.ts (hunk 9 above)
Comment (alessandrogelmi) [reply packages/pn-personafisica-webapp/src/redux/notification/reducers.ts:94]: Ok, but I can't use them here because I've to cast the pagoPa property

**Review by ZorbaDimatteo — APPROVED** [review]

**Pull request merged**